src/WeekCalendar/CalendarCells.tsx

```diff
@@ -2,28 +2,45 @@ import styles from './weekCalendar.module.css';
 import { DAYS_IN_WEEK, CELLS_IN_COLUMN } from '../constants';
 import { createArray } from '../utils';
 import { getCellDateForWeekCalendar } from '../dateUtils';
+import { Event } from '../types';
+import Cell from './Cell';
+
 interface CalendarCellsProps {
   calendarDate: Date;
   openModal: (date: Date) => void;
+  events: Event[];
 }
 
-const CalendarCells = ({ calendarDate, openModal }: CalendarCellsProps) => {
+const CalendarCells = ({
+  calendarDate,
+  openModal,
+  events,
+}: CalendarCellsProps) => {
   const handleCellClick = (date: Date) => {
     openModal(date);
   };
 
   return (
-    <div className={styles.calendarCells}>
+    <div className={styles.calendarCells} id="calendarCells">
       {createArray(DAYS_IN_WEEK * CELLS_IN_COLUMN).map((_, i) => {
         const cellDate = getCellDateForWeekCalendar(calendarDate, i);
         const key = `${calendarDate.getDate()}-${calendarDate.getHours() + i}`;
 
+        const eventsForCell = events?.filter((event) => {
```

Comment:
why optional chaining operator?

Reply:
also move to getEventsForCell function

Reply:
It errors, if I don't, because events can be undefined if there is no events in current week